### ISSUE TITLE
Ensure all ARIA attributes have valid values [accessibility audit #14]

### ISSUE THREAD

**Issue by jemnery** [ministryofjustice/find-moj-data#1512]
GDS Contact Email - [govuk-design-system-support@digital.cabinet-office.gov.uk](mailto:govuk-design-system-support@digital.cabinet-office.gov.uk).

### Describe the bug.

# Name, role, value

## ARIA attributes do not have valid values

On the User Guide page, the `#search` element uses an `aria-controls` attribute with a value of `"search-results"`, which does not reference a matching element `id` present in the DOM. This results in an invalid ARIA attribute value, which may cause confusion for users of assistive technologies.

ARIA attributes like `aria-controls` must point to a valid `id` of an element that exists on the page. When these relationships are broken or misused, assistive technologies may not be able to interpret or communicate these relationships accurately, which can reduce the semantic clarity of the interface.

At the time of testing, no issues were observed when interacting with the component using assistive technologies such as screen readers. However, the use of invalid ARIA attribute values is not considered best practice and could lead to issues in future updates or in different environments. Ensuring semantic accuracy helps maintain long-term accessibility and compliance.

![Search field highlighted on the user guide page](https://uvxxxx-testproject.s3.eu-west-2.amazonaws.com/SNAG_Program-0066.png)

### Code snippets
```
<div data-module="search" data-path-to-site-root="./">
  <form action="..." method="get" role="search" 
data-landmark-index="2">
    <label for="search">Search this documentation</label>
    <input type="search" id="search" name="q" 
 aria-controls="search-results" placeholder="Search">
    <button type="submit">Search</button>
  </form>
</div>
```

### Recommendation
Ensure that the `aria-controls` attribute references a valid `id` of an element present in the DOM. If the intention is to use `aria-controls` to create an accessible relationship between the search input and a results container, a corresponding element with the appropriate id should be added to the page. e.g. 

```
<input type="search" id="search" name="q" 
 aria-controls="search-results" placeholder="Search">

<!-- Add this somewhere appropriate on the page -->
<div id="search-results" aria-live="polite"></div>
```

If this relationship is not required, the attribute should be removed to avoid invalid markup. e.g. 

`<input type="search" id="search" name="q" placeholder="Search">`

**Comment by DeanWillicombe82** [comment]
This is a WCAG low category failure at present. @jemnery to check to see if this was implemented.

**Comment by jemnery** [comment]
Once a search is performed, the `search-results element` **does** exist, so this is only an issue on the home page. If we still want to address this, I think it would need to be raised as a bug with the team who controls the tech docs template (i.e. "don't show the `aria-controls` attribute until after a search is performed).

<img width="1524" alt="Image" src="https://github.com/user-attachments/assets/5bf6e416-21f1-4a92-a29a-4d071ed260d2" />

**Comment by DeanWillicombe82** [comment]
Hi Jeremy, How are you ?
We have a couple of tickets in relation to those accessibility issues, which you were looking to hunt down an owner of the code - [1512](https://github.com/ministryofjustice/find-moj-data/issues/1512) and [1514](https://github.com/ministryofjustice/find-moj-data/issues/1514), have you made any progress at all ?

**Comment by DeanWillicombe82** [comment]
This accessibility issue relates to the user guide, prior to migration into the main service which occured on 25/06/2025. Ticket can now be closed.
